Added support for opensuse 15.3 Cloud image

venkatnsrinivasan opened the pull request.

hxietkg committed:
Modify permission of SSH folder

SSH is installed in folder 'C:\Program Files\OpenSSH', according to
wiki from Win32-OpenSSH:
   https://github.com/PowerShell/Win32-OpenSSH/wiki/Install-Win32-OpenSSH
It needs the folder has correct permissions set. Now the folder inherits
users and permissions from its parent folder, SYSTEM and Administrators
have already had correct permissions, the only exception is that user
BUILTIN\Users has wrong permission, we are not able to connect to sshd
because of the wrong permission set, so let's just remove it.

jzhoucliqr commented on images/capi/packer/config/kubernetes.json:
```diff
@@ -12,9 +12,9 @@
-  "kubernetes_rpm_version": "1.21.9-0",
-  "kubernetes_semver": "v1.21.9",
-  "kubernetes_series": "v1.21",
+  "kubernetes_rpm_version": "1.20.14-0",
```
Is this change specific to us?

venkatnsrinivasan replied:
yes , incorrect checkin . need to back this out.

venkatnsrinivasan replied:
Reverted

jzhoucliqr commented on images/capi/ansible/roles/providers/tasks/suse.yml:
```diff
@@ -0,0 +1,19 @@
+  until: azure_cli_installation_rpm is success
+  retries: 15
+  delay: 60
+- name: install azure-cli
```
why this is needed?

venkatnsrinivasan replied:
Not particularly sure . Following template of redhat, debian tasks to be consistent ?

Commit:
capi: support building flatcar-gen2 images

Commit:
capi: make PATH extension work when ran from a different directory

When `make` is executed as `make -C images/cap`, PWD does not point to
images/capi, but to the directory where the user is located. Thus extending
PATH to find the packer fetched in 'ensure-packer.sh' was not working. Fix this
by using CURDIR instead of PWD, which will work in both cases.

Commit:
capi: override FLATCAR_VERSION to correct version when using "current"

There is subtle difference in how make handles variable overrides from env vs
passed as arguments to make (FLATCAR_VERSION=... make vs make
FLATCAR_VERSION=...). In the second case an 'override' is needed, otherwise
FLATCAR_VERSION stays with the value 'current'. This unifies behavior for both
cases and prevents nasty surprises.

venkatnsrinivasan committed:
Added support for opensuse 15.3 Cloud image

hxietkg committed:
Add ssh_source_url to all windows packer

codenrhoden committed:
Fix tdnf install in firstboot role

The tdnf command in the firstboot role is happening before any custom
repos are put in place. This means that if you happen to define a
private or internal mirror to use instead of a public one, the firstboot
role ignores this and uses the public repo anyways.

The logic in the setup role (meant to be used as a dependency) already
accounts for this, and needs to be used instead.

johananl committed:
Add missing comma in vsphere.md

k8s-ci-robot committed:
Merge pull request #852 from johananl/patch-1

Add missing comma in vsphere.md

jsturtevant committed:
Absolute in Windows kubelet don't work < 1.23

k8s-ci-robot committed:
Merge pull request #856 from jsturtevant/fix-kubelet-paths-windows

Absolute in Windows kubelet don't work < 1.23

k8s-ci-robot committed:
Merge pull request #851 from codenrhoden/photon-first-boot-repo

Fix tdnf install in firstboot role

SanikaGawhane committed:
Set transparent huge pages to madvise

johananl committed:
qemu: Fix Flatcar build

- Don't assume Kickstart boot over HTTP.
- Use password auth for Flatcar.
- Add Flatcar GOSS var.
- Allow overriding Flatcar installation wait time.
- Remove dedicated Flatcar build config.

invidian committed:
images/capi/packer/azure: pass image_version from user to sig builder

For VHDs, this is already done, so it should be done for SIG images as
well.

Refs #861
Refs #862

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

invidian committed:
images/capi/packer/azure/flatcar.json: specify image version

Right now SIG Flatcar images always use latest SKU image version,
despite user specifying requested version via FLATCAR_VERSION
environment variable.

This has been noticed due to regression in latest Flatcar stable
release, which right now broke OS detection and rules in CAPI Ansible
playbooks.

This commit correctly passes FLATCAR_VERSION environment variable into
image_version input variable for builders.

Refs #862

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

k8s-ci-robot committed:
Merge pull request #813 from hxietkg/modify_ssh_perm

Modify permission of SSH folder

seanschneeweiss committed:
containerd: update to v1.6.2

invidian committed:
images/capi/ansible/roles/setup: override ansible_os_family for Flatcar

Flatcar stable relase since 3139.2.0 has removed
/etc/flatcar/update.conf file from default instalation, which was used
by Ansible to determine the distro type and effectively the value of
'ansible_os_family' fact, which broke building Flatcar images, as now
the conditions for tasks does not evaluate properly when running on
Flatcar, as 'ansible_os_family' is reported as
'Flatcar Container Linux by Kinvolk' instead of just 'Flatcar'.

This commit adds a workaround for this issue by overriding the value for
'ansible_os_family' fact if problematic value is detected.

An alternative to this would be to set a custom fact named e.g.
'is_flatcar' and use that in code, which allows to deduplicate and
encapsulate the logic of determining if Flatcar is in use. However, once
Ansible bug is fixed on determining 'ansible_os_family' value, this code
can be removed and remaining code can stay as is.

See links below for more details:
- https://github.com/flatcar-linux/scripts/pull/212
- https://github.com/ansible/ansible/issues/77537
- https://github.com/kubernetes-sigs/image-builder/issues/862

Refs #862

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

k8s-ci-robot committed:
Merge pull request #864 from mercedes-benz/seanschneeweiss/containerd-v1.6.2

containerd: update to v1.6.2

codenrhoden committed:
Remove deprecated HAProxy OVA

CAPV has not needed the HAProxy OVA for a couple years now, and we had
long since announced the HAProxy as deprecated. This commit removes the
build target and associated config for building the HAProxy OVA.

invidian committed:
images/capi/ansible/roles/setup: remove redundant Flatcar condition

flatcar.yml is already only included on Flatcar, so this check can be
dropped.

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

invidian committed:
images/capi/ansible: add some comments on inconsistencies and config

I think it's useful to have those comments in place in case someone want
to change those line, like I wanted.

Also original commit or PR adding this code does not explain why this
options are needed.

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

k8s-ci-robot committed:
Merge pull request #863 from kinvolk/invidian/fix-azure-sig-image-version-flatcar

Fix building Flatcar images for Flatcar stable releases after 3139.2.0

Commit:
capi: bump default FLATCAR_VERSION to 3033.2.4

Flatcar added Azure Gen2 support in 2983.0.0. Let's bump the default to
3033.2.4, which was the last release with the 5.10.x kernel and also the base
of the coming lts-2022 Flatcar release.

Signed-off-by: Jeremi Piotrowski <jpiotrowski@microsoft.com>

k8s-ci-robot committed:
Merge pull request #829 from kinvolk/johananl/fix-flatcar-qemu

Fix QEMU Flatcar build

rymancl committed:
feat: expose security group and ssh variables in capi packer ami

- allow overriding security_group_ids
- allow overriding ssh_keypair_name
- allow overriding ssh_private_key_file

k8s-ci-robot committed:
Merge pull request #869 from codenrhoden/remove-haproxy

Remove deprecated HAProxy OVA

codenrhoden committed:
Separate OVA CI logs into separate files

This makes the OVA CI logs look like the Azure ones.

johananl committed:
ami: Use cloud-init only on supported distros

Not all Linux distros support cloud-init. Flatcar Container Linux,
for example, uses Ignition as a bootstrap format.

In #837, a cloud-init user data script was added for all distros.

- Move user_data to a variable to allow per-distro overrides.
- Set the default to the existing cloud-init script.
- Override to an empty script for Flatcar.

Signed-off-by: Johanan Liebermann <jliebermann@microsoft.com>

k8s-ci-robot committed:
Merge pull request #872 from codenrhoden/ova-ci-logs

Separate OVA CI logs into separate files

jsturtevant committed:
use com2 serial port for logging on Azure

ThatsMrTalbot committed:
Quote multi-value fields in ansible-args.json

When multiple space separated values are passed to fields such as extra_repos the intent is they all get actioned by ansible, but due to a lack of quoting in ansible-args.json only the first value gets passed in to ansible at all.

k8s-ci-robot committed:
Merge pull request #875 from jsturtevant/azure-serial-port

Use com2 serial port for logging on Azure

jsturtevant committed:
unpin WS 2022 image

jsturtevant committed:
use link with no default locale

Co-authored-by: Mark Rossetti <marosset@microsoft.com>

mboersma committed:
Update Azure scripts for new naming scheme

k8s-ci-robot committed:
Merge pull request #877 from jsturtevant/ws2022-node-port-fixes

unpin WS 2022 image

k8s-ci-robot committed:
Merge pull request #876 from ThatsMrTalbot/patch-1

Quote multi-value fields in ansible-args.json

k8s-ci-robot committed:
Merge pull request #874 from kinvolk/johananl/ami-user-data

ami: Use cloud-init only on supported distros

invidian committed:
Improve building Azure SIG images (#867)

* hack/ensure-ansible.sh: install collections in batch

So message "Nothing to do. All requested collections are already
installed. If you want to reinstall them, consider using" does not get
printed several times but only once on subsequent runs.

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

* packer/azure/scripts/init-sig.sh: remove trailing whitespace

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

* images/capi: create only built Azure SIG image definitions

Right now when building any of SIG images, all image definitions are
created, which makes it more time consuming to clean up when testing.

With this commit, only currently built definition will actually be
ensured to exist, which makes it easier to clean up and much faster, as
now running init-sig.sh for every SIG image built takes ~10 minutes,
instead of 1h.

Refs #855

Signed-off-by: Mateusz Gozdek <mgozdekof@gmail.com>

k8s-ci-robot committed:
Merge pull request #871 from rymancl/capi-packer-ami-json-additions

feat: expose security group and ssh variables in capi packer ami

johananl committed:
Simplify conditional in Flatcar provisioner

BUILD_NAME is already set in environment_vars. No need to set it again
inline.

Signed-off-by: Johanan Liebermann <jliebermann@microsoft.com>

johananl committed:
qemu: Set Flatcar boot_media_path

Flatcar uses Ignition rather than Kickstart.

The build isn't broken because apparently the boot_media_path part of
the boot command is typed after the reboot command is executed,
however we should set this var to an empty string to be safe.

Signed-off-by: Johanan Liebermann <jliebermann@microsoft.com>

johananl committed:
qemu: Remove unnecessary boot_media_path overrides

boot_media_path has a default in place. Distro-specific values should
be set for this var only when the default is wrong.

Signed-off-by: Johanan Liebermann <jliebermann@microsoft.com>

k8s-ci-robot committed:
Merge pull request #880 from kinvolk/johananl/fix-qemu-boot-media-path

Fix QEMU boot_media_path overrides

codenrhoden committed:
ova:Make sure goss inpsect mode is disabled for CI

k8s-ci-robot committed:
Merge pull request #868 from SanikaGawhane/sg-THP-madvise

Set transparent huge pages to madvise

k8s-ci-robot committed:
Merge pull request #883 from kinvolk/johananl/align-flatcar-provisioner

Simplify conditional in Flatcar provisioner

k8s-ci-robot committed:
Merge pull request #884 from codenrhoden/ova-goss-inspect

ova:Make sure goss inpsect mode is disabled for CI

knabben committed:
Enabling custom roles instructions

cpanato committed:
remove nightly image fo 1.20

Signed-off-by: cpanato <ctadeu@gmail.com>

cpanato committed:
add nightly for 1.24

Signed-off-by: cpanato <ctadeu@gmail.com>

cpanato committed:
update 1.23/1.22/1.21 nightly images

Signed-off-by: cpanato <ctadeu@gmail.com>

k8s-ci-robot committed:
Merge pull request #886 from knabben/custom-role-docs

Enabling custom roles instructions

k8s-ci-robot committed:
Merge pull request #887 from cpanato/update-gce-images

Update gce images

johananl committed:
Revert "Simplify conditional in Flatcar provisioner"

This reverts commit 4fb09d311a139aa67ded9c7cdcc3f8d35bd50c07.

johananl committed:
Clean up Flatcar provisioner

Since we already check the build name inside execute_command, there
is no need to check the same again inside bootstrap-flatcar.sh.
Moreover, having both conditionals is confusing and error-prone.

Putting the conditional in the Packer config is more visible than
doing so inside the script. It's arguably also more intuitive, since
now bootstrap-flatcar.sh is only executed when the build is a Flatcar
build (as opposed to always executing and then exiting when the build
isn't Flatcar).

So, we keep the conditional in execute_command and remove the one in
bootstrap-flatcar.sh.

- Remove the conditional inside bootstrap-flatcar.sh.
- Remove environment_vars since it affects only bootstrap-flatcar.sh
  and has no influence on execute_command.

Signed-off-by: Johanan Liebermann <jliebermann@microsoft.com>

codenrhoden committed:
ova: remove downgrade of cloud-init

With cloud-init v22.1 available, no longer a need to downgrade on photon
and Ubuntu. EL distros are still not >- 21.3, so they still need to the
VMware Guestinfo Datasource.

codenrhoden committed:
refactor role providers/vmware

A lot of distro-specific logic was in main.yml, while we already had a
couple of vmware-distro.yml files to be included. this patch adds a
vmware-redhat.yml file, and makes sure all vmware-distro files are
included by vmware.yml. This refactoring makes it much easier to see
what logic is running under what circumstances.

k8s-ci-robot committed:
Merge pull request #889 from codenrhoden/cloud-init

ova: Remove downgrade of cloud-init, no longer install guestinfo datasource when not needed

k8s-ci-robot committed:
Merge pull request #888 from kinvolk/johananl/fix-bootstrap-flatcar

Fix regression in Flatcar provisioner

k8s-ci-robot committed:
Merge pull request #878 from mboersma/update-azure-offer

Update Azure scripts for new naming scheme

johananl committed:
qemu,raw: Mask sshd during bootstrap

Flatcar ships with sshd enabled by default. Installing Flatcar
entails booting into Flatcar from an ISO and running the installation
command. Packer treats the availability of the SSH port as an
indication that the OS installation is done. To address this, a
static wait time is appended to the Flatcar boot command, which means
there is a potential race between the installation script and Packer.
This leads to occasional Packer errors in Flatcar builds.

Mask sshd during bootstrap. When the installation completes, the VM
is rebooted from disk and then sshd is enabled again automatically.
Also remove the wait as it's no longer necessary because we can rely
on the built-in ssh_timeout Packer parameter like other distros do.

Signed-off-by: Johanan Liebermann <jliebermann@microsoft.com>

kkeshavamurthy committed:
Modify cloud.cfg to preserve hostname on reboot for OVAs

kkeshavamurthy committed:
Bump k8s version v1.22.9

k8s-ci-robot committed:
Merge pull request #891 from kkeshavamurthy/bump-k8s-1.22

Bump k8s version to v1.22.9

mboersma committed:
Remove KUBERNETES_VERSION from Azure SKU template

k8s-ci-robot committed:
Merge pull request #896 from mboersma/update-azure-sku-template

Remove KUBERNETES_VERSION from Azure SKU template

k8s-ci-robot committed:
Merge pull request #893 from kinvolk/johananl/flatcar-mask-sshd

qemu,raw: Mask sshd during bootstrap

k8s-ci-robot committed:
Merge pull request #834 from kinvolk/jepio/flatcar-gen2

flatcar Azure gen2 images

kkeshavamurthy committed:
tune node images for max etcd network performance

Pearl1594 committed:
Update to Rocky Linux 8.6

k8s-ci-robot committed:
Merge pull request #900 from kkeshavamurthy/kk-hostname

Modify cloud.cfg to preserve hostname on reboot for OVAs

k8s-ci-robot committed:
Merge pull request #901 from kkeshavamurthy/kk-etcd-rules

tune node images for max etcd network performance

k8s-ci-robot committed:
Merge pull request #899 from shapeblue/update-r8-link

Update to Rocky Linux 8.6

mkumatag committed:
Add IBMCloud PowerVS support (#860)

* Add PowerVS support

* fix comments

* fix the ci error

* fix crictl checksum

davidjumani committed:
Adding support for CloudStack capi images

cpanato committed:
build 1.21.13/1.22.10/1.23.7/1.24.1 nightly images

Signed-off-by: cpanato <ctadeu@gmail.com>

k8s-ci-robot committed:
Merge pull request #909 from cpanato/update-gce-images

build 1.21.13/1.22.10/1.23.7/1.24.1 nightly images

k8s-ci-robot committed:
Merge pull request #842 from shapeblue/add-cloudstack-support

Adding support for CloudStack capi images
